.docker/Dockerfile

```
@@ -0,0 +1,120 @@
+		echo 'html_errors = Off'; \
+	} > /usr/local/etc/php/conf.d/error-logging.ini
+
+# ----- Copied section ends ----- #
+
+RUN apt-get update
+
+# Download WordPress source code. We are downloading from git so that we can checkout different WordPress versions by
+# just checking out the branch.
+RUN apt-get install git -y
+RUN mkdir -p /usr/src/public_html/ && cd /usr/src/public_html/ && git clone git://github.com/WordPress/WordPress.git mu
+RUN cd /usr/src/public_html/mu && git checkout 5.2-branch
+WORKDIR /usr/src/public_html
+
+# Install and configure nginx
+RUN apt-get install nginx -y
+COPY nginx.conf /etc/nginx/sites-enabled/wordcamp.conf
+COPY php-fpm.conf /usr/local/etc/php-fpm.d/zz-www.conf
+
+# Install and configure SSL
+COPY wordcamp.ssl.conf /var/wordcamp.ssl.conf
+RUN cd ~ && openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout wordcamp.key -out wordcamp.crt -config /var/wordcamp.ssl.conf
+RUN cp ~/wordcamp.crt /etc/ssl/certs/wordcamp.crt
+RUN cp ~/wordcamp.key /etc/ssl/private/wordcamp.key
+
+# Add script to install plugin and themes
+ADD install-plugin-theme.sh /usr/src/public_html/install-plugin-theme.sh
+
+# Unfortunately themes and plugins cannot be installed as part of docker setup and have to be run manually.
+# This is because everything in wp-content folder will be overwritten when we mount this directory from volume.
+# RUN sh install-plugin-theme.sh
+
+
+# Install `wkhtmltopdf` for WordCamp Docs, CampTix Invoices, etc. See https://stackoverflow.com/a/38336153/1845153
+RUN curl -L https://github.com/wkhtmltopdf/wkhtmltopdf/releases/download/0.12.4/wkhtmltox-0.12.4_linux-generic-amd64.tar.xz > wkhtmltox.tar.xz
+RUN apt-get install libxext6 -y
+RUN tar xvf wkhtmltox.tar.xz
+RUN mv wkhtmltox/bin/wkhtmlto* /usr/bin/
+RUN rm -rf wkhtmltox*
+
+# Install WP-CLI
+RUN curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+RUN chmod +x wp-cli.phar
+RUN mv wp-cli.phar /usr/local/bin/wp
+RUN apt-get install less -y
+COPY wp-cli.yml /usr/src/public_html/wp-cli.yml
+
+# Install MailCatcher.
+RUN apt-get install build-essential libsqlite3-dev ruby-dev -y
+RUN gem install mailcatcher --no-ri --no-rdoc
+RUN apt-get install ssmtp -y
+RUN sed -i -e "s|;sendmail_path =|sendmail_path = /usr/sbin/ssmtp -t |" /usr/local/etc/php/php.ini-development
+RUN sed -i -e "s/smtp_port = 25/smtp_port = 1025/" /usr/local/etc/php/php.ini-development
+RUN chown root:mail /etc/ssmtp/ssmtp.conf
+RUN sed -i -e "s/#FromLineOverride=YES/FromLineOverride=YES/" /etc/ssmtp/ssmtp.conf
+RUN sed -i -e "s/mailhub=mail/mailhub=127.0.0.1:1025/" /etc/ssmtp/ssmtp.conf
+RUN cp /usr/local/etc/php/php.ini-development /usr/local/etc/php/php.ini
+
+# Utils
+RUN apt-get install vim -y
+
+COPY wp-config.php /usr/src/public_html/wp-config.php
+
+# Replace nginx default conf because it conflicts with wordcamp.cong
+RUN mv /etc/nginx/sites-enabled/wordcamp.conf /etc/nginx/sites-enabled/default
+
+CMD tail -f /dev/null
+
```

```suggestion

RUN echo "\n\nSuccess: The WordCamp container is now running.\n"
RUN echo "You can visit https://2014.content.wordcamp.org/ to view a sample WordCamp site. You can log in with username 'admin' and password 'password'.\n"
RUN echo "You can also visit http://localhost:1080/ to open MailCatcher."

```

`MUST` - At first I got tripped up because I didn't realize that `docker-compose` up was supposed to run indefinitely. I expected it to finish and return me to the shell, similar to a `vagrant up`, but it just sat there after starting MySQL for several minutes before I got it. This will help avoid that, and also give the user some necessary information.

Alternatively, it'd be cool if instead of showing the Nginx access log, if the terminal instead automatically entered a shell inside the container, so that you don't have to do that manually. If we don't do that, we'll probably want a quick way to enter it without having to lookup the current name and type a command. Maybe just `ssh wordcamp.test` via `~/.ssh/config` or something?

---

I have edited Docker's `readme.md` and changed some of the provisioning command.
`docker-compose up --build -d` will exit when provisioning is done and docker instance will keep on running in background. 

Also `docker-compose exec wordcamp.test bash` can be used to go inside the shell. See https://github.com/WordPress/wordcamp.org/pull/113/commits/ac11b763700e157d761baedda4253bf3722d6e60

Please resolve the conversation if you feel that this addressed your issues.

---

That commit looks great!